### DISCUSSION TITLE
Potential issue with Pytorch MIONet when using input transform

### DISCUSSION THREAD

**Discussion by Dkupetsk** [lululxvi/deepxde#1542] [Q&A] [answered]
Hello,

I encountered an error in my project when I tried to apply a feature transform in MIONet:

"RuntimeError: mat1 and mat2 shapes cannot be multiplied (1000x1 and 6x100)"

My feature transform was supposed to send my trunk network input to a MxNx6 array. I noticed that in the MIONet PyTorch the code says the following:

```
if self._input_transform is not None:
    y_loc = self._input_transform(x_loc)
y_loc = self.trunk(x_loc)
```

After changing this to 

```
if self._input_transform is not None:
    x_loc = self._input_transform(x_loc)
y_loc = self.trunk(x_loc)
```

I no longer had any issues. Is this a potential typo?

Thanks

**Comment by lululxvi** [comment]
The code looks correct.

https://github.com/lululxvi/deepxde/blob/26860b17d90d7c72dd1d70210941207ff408b312/deepxde/nn/pytorch/deeponet.py#L52

Can you point out the code?

**Reply by Dkupetsk** [reply]
Yeah it looks fine in DeepONet, but MIONet is different:
[see here](https://github.com/lululxvi/deepxde/blob/26860b17d90d7c72dd1d70210941207ff408b312/deepxde/nn/pytorch/mionet.py#L112C1-L114C34)
```
if self._input_transform is not None:
            y_loc = self._input_transform(x_loc)
        y_loc = self.trunk(x_loc)

```

shouldn't it say 

```
if self._input_transform is not None:
            x_loc = self._input_transform(x_loc)
        y_loc = self.trunk(x_loc)

```

**Reply by lululxvi** [reply, marked as answer]
Yes, you are right. It is a bug. Do you want to submit a PR to fix this?

**Reply by Dkupetsk** [reply]
Sure!
